GitUI/CommandsDialogs/FormArchive.cs

```diff
@@ -160,7 +160,8 @@ private string GetPathArgumentFromGui()
             else if (checkboxRevisionFilter.Checked)
             {
                 // 1. get all changed (and not deleted files) from selected to current revision
-                var files = UICommands.Module.GetDiffFiles(DiffSelectedRevision.Guid, SelectedRevision.Guid, SelectedRevision.ParentIds.FirstOrDefault()?.ToString()).Where(f => !f.IsDeleted);
+                var stagedStatus = GitCommandHelpers.GetStagedStatus(DiffSelectedRevision?.ObjectId, SelectedRevision?.ObjectId, DiffSelectedRevision?.ObjectId);
```

Comment:
In a pure refactoring, I had expected:
```suggestion
                var stagedStatus = GitCommandHelpers.GetStagedStatus(DiffSelectedRevision?.ObjectId, SelectedRevision?.ObjectId, SelectedRevision?.ParentIds?.FirstOrDefault());
```

Reply:
I believe I had doubt if parents were set for the revision, but DiffSelectedRevision is not always the parent so this could be incorrect
However, comparisions cannot be done to artificial, so StagedStatus is always None

Rebasing and squashing this